### PR TITLE
Remove ICONV dep

### PR DESCRIPTION
**Pull request by rozhuk-im**

### CODE REVIEW

**Comment by RodneyBaker** [comment]
This is looking good so far but I am in early stage of testing as much of this is unfamiliar territory.
Just wanted you to know someone else is looking into the PR @rozhuk-im.

File: toonz/sources/CMakeLists.txt [hunk 1]
```diff
@@ -458,6 +458,8 @@ elseif(BUILD_ENV_UNIXLIKE)
     find_package(LZO REQUIRED)
     message("LZO:" ${LZO_INCLUDE_DIR})
 
+    find_library(PTHREAD_LIBRARY pthread)
```
Comment (martinvanzijl) [inline toonz/sources/CMakeLists.txt:461]: Why is pthread needed?

What is the risk the user does not have pthread installed?

**Comment by martinvanzijl** [comment]
Overall this looks OK. To clarify the changes:

- Iconv dependency removed.
- Pthread dependency added.
- Some unused functions removed, .e.g `locale_to_jp`.
- Some function declarations removed from header files, since they are not used externally, e.g. `mbs_to_wcs`.
- In igs_resource_msg_from_err_win.cpp, in `std::string mbs_from_ts`, the code-page parameter is now hard coded to `CP_THREAD_ACP` (this was the default value). Before the change it could also be either `CP_UTF8` (from `utf8_from_cp932_mb`) or `932` (from `cp932_from_utf8_mb`) but these functions have both been removed.

The affected code seems to be called from the "Fog Ino" effect, if there is an error in one of its threads. I'm not sure how to trigger this, though. 

Compiles and runs OK for me on Ubuntu 18.04. This should also be compiled/tested on Windows and Mac. I recommend testing an animation with the "Fog Ino" effect and seeing it still works OK.

**Comment by martinvanzijl** [comment]
@shun-iwasawa Do you happen to know who wrote the "Fog Ino" effect? It would be nice to check with them too.

File: toonz/sources/CMakeLists.txt (hunk 1 above)
Comment (rozhuk-im) [reply toonz/sources/CMakeLists.txt:461]: This is required it app uses threads, ```#include <pthread.h>``` + any ```pthread_*()``` requires to add this.
On linux and probably mac pthread is somewhere in standard lib, IMHO, that is why this work before. But on BSD pthread requires additional lib for linker.

**Comment by shun-iwasawa** [comment]
@martinvanzijl 
> Do you happen to know who wrote the "Fog Ino" effect?

The related codes were written by @masafumi-inoue . 

@rozhuk-im 
Thank you for your modifications for FreeBSD support. 
I sought a comment from @masafumi-inoue regarding this PR. Unfortunately he doesn't have enough time for reviewing for now. He is not sure if there would be no problem by removing the iconv-related codes as he wrote it a long time ago. 
He's just wondering if, isn't it an option that installing libiconv to FreeBSD instead of removing dependency.

**Comment by rozhuk-im** [comment]
@shun-iwasawa This is not FreeBSD related code. Linux users also affected: https://github.com/opentoonz/opentoonz/issues/2467

FreeBSD have iconv lib, and there was 2 options to build OpenToonz:
1. Add -liconv to CMake file
2. Check is it really used and remove if not.

I found that this code does not used, you may recheck this to be sure that there is no mistake.
IMHO this is artifact from pre utf-8 times. :)

**Comment by shun-iwasawa** [comment]
Jenkins

**Comment by shun-iwasawa** [comment]
Jenkins

**Comment by shun-iwasawa** [comment]
@masafumi-inoue reviewed and verified the changes.
I also confirmed that it works with no problem both on Windows and MacOS.

LGTM Thank you @rozhuk-im

**Comment by rozhuk-im** [comment]
https://github.com/opentoonz/opentoonz/issues/3597

**Pull request merged**